src/neo4j_graphrag/experimental/components/entity_relation_extractor.py

```diff
@@ -153,17 +152,19 @@ async def run(
         pass
 
     def update_ids(
-        self, graph: Neo4jGraph, chunk_index: int, run_id: str
+        self,
+        graph: Neo4jGraph,
+        chunk: TextChunk,
     ) -> Neo4jGraph:
-        """Make node IDs unique across chunks and pipeline runs by
-        prefixing them with a custom prefix (set in the run method)
-        and chunk index."""
-        prefix = f"{run_id}:{chunk_index}"
+        """Make node IDs unique across chunks, document and pipeline runs
+        by prefixing them with a unique prefix.
+        """
+        prefix = f"{chunk.chunk_id}"
```

Comment:
If you remove the `run_id` from the node ID will this not cause an issue if you run a pipeline multiple times over the same document? Or is the idea that it should fail in that situation?

Reply:
Now the `chunk_id` is unique (it's a UUID), even if we rerun the pipeline on the same document (previously, we were using the chunk index as chunk ID, this is why we needed a prefix to make the entity IDs unique, but this was the cause of the bug: the chunk IDs were not unique, and so rerunning the pipeline multiple times on the same doc was messing up the relationships between chunks)

Reply:
I see. So, all else being equal, would two runs over the same document produce two graphs with no links between them?

Reply:
Yes